src/charm.py

```diff
@@ -169,6 +173,11 @@ def _restart_flask(self) -> None:
         except CharmConfigInvalidError as exc:
             self._update_app_and_unit_status(ops.BlockedStatus(exc.msg))
 
+    def _on_update_status(self, _: ops.HookEvent) -> None:
+        """Handle the update-status event."""
+        if self._database_migration.get_status() == self._database_migration.FAILED:
```

Comment:
Should `self._database_migration.FAILED` just be a constant rather than a property of the class